api/include/opentelemetry/trace/trace_state.h

```diff
@@ -45,51 +45,146 @@ class TraceState
   class Entry
   {
   public:
-    Entry() noexcept = default;
-
-    // Copy constructor.
-    Entry(const Entry &copy);
+    Entry() : key_(nullptr), value_(nullptr){};
+
+    // Copy constructor
+    Entry(Entry &copy)
+    {
+      key_   = CopyStringToPointer(copy.key_.get());
+      value_ = CopyStringToPointer(copy.value_.get());
+    }
+
+    // Assignment operator
+    Entry &operator=(Entry &other)
+    {
+      key_   = CopyStringToPointer(other.key_.get());
+      value_ = CopyStringToPointer(other.value_.get());
+      return *this;
+    }
```

Comment:
Since we explicitly define copy ctor and assigment operator here, it worth to also provide move ctor and move assignment operator that would simply move key_ and value_ by calling move ctor and move assignment operator for nostd::unique_ptr

I think it would be sufficient to just add

Entry(Entry&& other) = default;
Entry& operator=(Entry&& other) = default;

As it would invoke per member move ctors, and currently default move ctor is not generated because of explicit copy ctor.